Objects/exceptions.c

```diff
@@ -216,6 +219,33 @@ BaseException_set_args(PyBaseExceptionObject *self, PyObject *val, void *Py_UNUS
     return 0;
 }
 
+static PyObject *
+BaseException_get_note(PyBaseExceptionObject *self, void *Py_UNUSED(ignored))
+{
+    if (self->note == NULL) {
+        Py_RETURN_NONE;
+    }
+    return Py_NewRef(self->note);
+}
+
+static int
+BaseException_set_note(PyBaseExceptionObject *self, PyObject *note,
+                       void *Py_UNUSED(ignored))
+{
+    if (note == NULL) {
+        PyErr_SetString(PyExc_TypeError, "__note__ may not be deleted");
+        return -1;
+    }
+    else if (note != Py_None && !PyUnicode_Check(note)) {
```

Comment:
As per C API docs https://docs.python.org/3/c-api/unicode.html#c.PyUnicode_Check I think this would allow subclasses of str so this would be https://docs.python.org/3/c-api/unicode.html#c.PyUnicode_CheckExact


```suggestion
    else if (note != Py_None && !PyUnicode_CheckExact(note)) {
```